src/liger_kernel/transformers/model/mixtral.py

```diff
@@ -0,0 +1,159 @@
+from typing import List, Optional, Tuple, Union
+
+import torch
+from torch.nn import CrossEntropyLoss
+from transformers.modeling_outputs import MoeCausalLMOutputWithPast
+from transformers.models.mixtral.modeling_mixtral import (
+    _CONFIG_FOR_DOC,
+    MIXTRAL_INPUTS_DOCSTRING,
+    load_balancing_loss_func,
+)
+from transformers.utils import (
+    add_start_docstrings_to_model_forward,
+    replace_return_docstrings,
+)
+
+from liger_kernel.transformers.fused_linear_cross_entropy import (
+    LigerFusedLinearCrossEntropyLoss,
+)
+
+
+@add_start_docstrings_to_model_forward(MIXTRAL_INPUTS_DOCSTRING)
+@replace_return_docstrings(
+    output_type=MoeCausalLMOutputWithPast, config_class=_CONFIG_FOR_DOC
+)
+# Ignore copy
```

Comment:
what is this

Reply:
If you're mentioning ignore copy, it's directly from transformers, should we remove it?

Reply:
done.